src/upgrade/check.rs

```diff
@@ -3,24 +3,60 @@ use ferinth::structures::version::{Version, VersionFile};
 use furse::structures::file_structs::File;
 use octocrab::models::repos::{Asset, Release};
 
+fn is_jar_file(asset_name: &str) -> bool {
+    asset_name.contains("jar")
+}
+
+fn is_not_source(asset_name: &str) -> bool {
+    !asset_name.contains("source")
+}
+
+fn game_version_check_contain(game_version: Option<&str>, asset_name: &str) -> bool {
+    game_version
+        .map(|game_version| asset_name.contains(game_version))
+        // select latest asset if version check is disabled
+        .unwrap_or(true)
+}
+
+fn game_version_check_exact(game_version: Option<&str>, version: &str) -> bool {
+    game_version
+        .map(|game_version| version == game_version)
+        // select latest asset if version check is disabled
+        .unwrap_or(true)
+}
+
+fn mod_loader_check_contain(mod_loader: Option<&ModLoader>, asset_name: &str) -> bool {
+    mod_loader
+        .map(|mod_loader| {
+            asset_name
+                .split('-')
+                .any(|loader| loader == mod_loader.to_string().as_str())
+        })
+        // select latest asset if mod loader check is disabled
+        .unwrap_or(true)
+}
+
+fn mod_loader_check_exact(mod_loader_to_check: Option<&ModLoader>, mod_loader: &str) -> bool {
```

Comment:
Another bug is caused by something I mentioned in my previous review, converting the user setting to a string and comparing doesn't work because when comparing strings you have to make sure they're the same case. Instead, it's better to try and parse the value from the server and _compare the enum_ instead (which is what was previously implemented)

Reply:
```patch
diff --git a/src/upgrade/check.rs b/src/upgrade/check.rs
index 83c61da..8654685 100644
--- a/src/upgrade/check.rs
+++ b/src/upgrade/check.rs
@@ -30,7 +30,7 @@ fn mod_loader_check_contain(mod_loader: Option<&ModLoader>, asset_name: &str) ->
         .map(|mod_loader| {
             asset_name
                 .split('-')
-                .any(|loader| loader == mod_loader.to_string().as_str())
+                .any(|loader| loader.parse().as_ref() == Ok(mod_loader))
         })
         // select latest asset if mod loader check is disabled
         .unwrap_or(true)
@@ -38,7 +38,7 @@ fn mod_loader_check_contain(mod_loader: Option<&ModLoader>, asset_name: &str) ->
 
 fn mod_loader_check_exact(mod_loader_to_check: Option<&ModLoader>, mod_loader: &str) -> bool {
     mod_loader_to_check
-        .map(|loader| loader.to_string().as_str() == mod_loader)
+        .map(|loader| mod_loader.parse().as_ref() == Ok(loader))
         // select latest asset if mod loader check is disabled
         .unwrap_or(true)
 }
```